.github/workflows/aws_deploy.yml

```diff
@@ -129,24 +129,46 @@ jobs:
           succeeded=$(aws ecs describe-tasks --cluster $cluster --tasks $id --query "tasks[0].stopCode == 'EssentialContainerExited' && to_string(tasks[0].containers[0].exitCode) == '0'")
           if [ $succeeded == true ]; then exit 0; else exit 1; fi
 
-      - name: Download service task definition
+      - name: Download app service task definition
         env:
           app_task_definition: ${{ inputs.aws_resource_prefix }}-app
         run: |
           aws ecs describe-task-definition --task-definition $app_task_definition --query taskDefinition > app-task-definition.json
 
-      - name: Update image ID
+      - name: Download sidekiq service task definition
+        env:
+          app_task_definition: ${{ inputs.aws_resource_prefix }}-sidekiq
+        run: |
+          aws ecs describe-task-definition --task-definition $app_task_definition --query taskDefinition > sidekiq-task-definition.json
+
+      - name: Update app image ID
         id: app-task-def
         uses: aws-actions/amazon-ecs-render-task-definition@v1
         with:
           task-definition: app-task-definition.json
           container-name: app
           image: ${{ env.image }}
 
+      - name: Update sidekiq image ID
+        id: sidekiq-task-def
+        uses: aws-actions/amazon-ecs-render-task-definition@v1
+        with:
+          task-definition: sidekiq-task-definition.json
+          container-name: sidekiq
+          image: ${{ env.image }}
+
       - name: Deploy updated application
         uses: aws-actions/amazon-ecs-deploy-task-definition@v1
         with:
           cluster: ${{ inputs.aws_resource_prefix }}
-          service: ${{ inputs.aws_resource_prefix }}
+          service: ${{ inputs.aws_resource_prefix }}-app
           task-definition: ${{ steps.app-task-def.outputs.task-definition }}
           wait-for-service-stability: true
+
+      - name: Deploy updated sidekiq
+        uses: aws-actions/amazon-ecs-deploy-task-definition@v1
+        with:
+          cluster: ${{ inputs.aws_resource_prefix }}
+          service: ${{ inputs.aws_resource_prefix }}-sidekiq
+          task-definition: ${{ steps.sidekiq-task-def.outputs.task-definition }}
+          wait-for-service-stability: true
```

Comment:
If this step fails, we then need manual intervention of some sort (whether that's re-running or rolling back) to ensure that sidekiq and the app are running the same version. This is fine, but when it comes to writing the runbook we should ensure that it's a documented thing to handle.

Reply:
@Chirag-Bhatti maybe add a placeholder in the runbook now to ensure it doesn't get forgotten?

Reply:
Sorry that sounded quite micromanaging...

Reply:
No problem! I've added a placeholder in the runbook [here](https://dluhcdigital.atlassian.net/wiki/spaces/MC/pages/32441141/PLACEHOLDER+Handling+incomplete+deployments+from+the+app+pipeline) with some very pre-emptive text :-)